src/components/RichEditor/index.jsx

```diff
@@ -21,9 +20,32 @@ class RichEditor extends React.Component {
     super(props);
     this.state = {
       charCount: 0,
+      value: props.input.value,
     };
     this.initCharCount = this.initCharCount.bind(this);
     this.updateCharCount = this.updateCharCount.bind(this);
+    this.handleEditorChange = this.handleEditorChange.bind(this);
+    this.handleOnChange = this.handleOnChange.bind(this);
+  }
+
+  componentDidUpdate(prevProps) {
+    const {
+      input: {
+        value,
+      },
+    } = this.props;
+
+    if (value !== prevProps.input.value) {
+      this.setState({ value });
```

Comment:
Is there no need to preserve the rest of the `state` (charCount) here?

Reply:
value state is set explicitly based on props (input.value), charCount is not. I did not observe any issue with charCount not being defined here.